nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,178 @@
+                type = lib.types.str;
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:7331";
+                description = "Server address.";
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Limit cookies to secure channel.";
+              };
+            };
+            storage = {
+              root = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."omnom/config.yml".source = pkgs.writers.writeYAML "omnom-config.yml" cfg.settings;
+
+    services.omnom.settings = {
+      app.debug = lib.mkDefault cfg.debug;
+      storage.type = lib.mkDefault "fs";
+    };
+
+    systemd.services.omnom = {
+      path = [ pkgs.rsync ];
+
+      serviceConfig = {
+        DynamicUser = isDynamicUser;
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart =
+          let
+            args = lib.cli.toGNUCommandLineShell { } {
+              config = "/etc/omnom/config.yml";
+              d = cfg.debug;
+            };
+          in
+          "${lib.getExe cfg.package} listen ${args}";
+        StateDirectory = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "omnom";
+        StateDirectoryMode = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "0750";
+        WorkingDirectory = cfg.dataDir;
+        Restart = "on-failure";
+        RestartSec = "10s";
+      };
+      after = [
+        "network.target"
+        "systemd-tmpfiles-setup.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      # These need to be present in the working directory
+      preStart = ''
+        for dir in templates; do
+          if [ ! -e "${cfg.dataDir}"/$dir ]; then
+            rsync -a --chmod=u=rwX,go=rX "${cfg.package}/share/$dir" "${cfg.dataDir}/"
+          fi
+        done
+      '';
+    };
+
+    # If the service user is not dynamic, a normal group must exist
+    users.groups = lib.mkIf ((!isDynamicUser) && (cfg.group == "omnom")) { omnom = { }; };
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/omnom") (
+      [
+        "d ${cfg.dataDir}                       0750 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.dataDir}/static                0750 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.dataDir}/static/data           0750 ${cfg.user} ${cfg.group} - -"
```

Comment:
a) Try using `systemd.tmpfiles.settings` instead and b) I *think* tmpfiles.d implicitly creates parent directories

Reply:
It's simpler and much more intuitive with named arguments instead of having to remember positional ones. Plus it's an attrset instead of a list which makes it more convenient in some cases

Reply:
@pluiedev Does [this](https://github.com/NixOS/nixpkgs/pull/357830/files#diff-bea2407e6c4faab93ae203e0d057587a28fcb3dd03e2f5b288b7709d7286b15aR157-R174) look good?